concent_api/core/tests/test_core_views.py

```diff
@@ -0,0 +1,208 @@
+        self.assertEqual(len(Message.objects.all()), 0)
+        self.assertEqual(len(MessageStatus.objects.all()), 0)
+
+        self.correct_data['message_task_to_compute']['deadline'] = self.message_timestamp - 1
+        response = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+
+        self.assertEqual(response.status_code, 200)  # pylint: disable=no-member
+        self.assertEqual(response.json()['type'], 'MessageRejectReportComputedTask')  # pylint: disable=no-member
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_data_is_incorrect(self):
+        data = {
+            "type":      "MessageForceReportComputedTask",
+            "timestamp": 1510911047,
+            "message_task_to_compute":      {
+                "type": "MessageTaskToCompute"
+            }
+        }
+
+        response = self.client.post(reverse('core:send'), data = json.dumps(data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+
+        self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
+        self.assertIn('error', response.json().keys())
+
+        self.correct_data['message_task_to_compute']['deadline'] = "1510909200"
+        response = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+
+        self.assertEqual(response.status_code, 400)  # pylint: disable=no-member
+        self.assertIn('error', response.json().keys())
+
+    @freeze_time("2017-11-17 10:00:00")
+    def test_send_should_return_http_400_if_task_id_already_use(self):
+        response_202 = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+
+        self.assertIsInstance(response_202, HttpResponse)
+        self.assertEqual(response_202.status_code, 202)
+
+        response_400 = self.client.post(reverse('core:send'), data = json.dumps(self.correct_data), content_type = 'application/json', concent_client_public_key = self.public_key, HTTP_CONCENT_CLIENT_PUBLIC_KEY = self.public_key)
+
+        self.assertIsInstance(response_400, JsonResponse)
+        self.assertEqual(response_400.status_code, 400)  # pylint: disable=no-member
+        self.assertIn('error', response_400.json().keys())
+
+
+class CoreViewReceiveTest(TestCase):
+
+    def setUp(self):
+        self.client = Client()
+        self.public_key = '85cZzVjahnRpUBwm0zlNnqTdYom1LF1P1WNShLg17cmhN2UssnPrCjHKTi5susO3wrr/q07eswumbL82b4HgOw=='
+        self.force_data = {
```

Comment:
Proponuję dodać jakieś abstract modele służące do generowanie tych struktur, model ten mógłby posiadać pola i metodę np. to_json() zwracającą tą strukturę, wtedy w testach moglibyśmy skorzystać z factory-boy do tworzenia danych.

Reply:
Our data won't be JSON. We need to work with golem-messages. This is not a good moment to refactor this since it's being rewritten now.

Reply:
:+1: